### PR TITLE
Fail lambda if shouldFail flag provided

### PR DESCRIPTION
**Pull request by sdomagala**
Hello,
It seemed to me that checking if each secret is available at start of lambda function is a bit too much if you are in situation where missing secrets mean there is no point for lambda to run.

### CODE REVIEW

File: src/middlewares/__tests__/secretsManager.js [hunk 1]
```diff
@@ -52,6 +52,7 @@ describe('🔒 SecretsManager Middleware', () => {
       promise = promise.then(() => {
         return new Promise((resolve, reject) => {
           handler(event, context, (error, response) => {
+            if (error) reject(error)
```
Comment (vladgolubev) [inline src/middlewares/__tests__/secretsManager.js]: Should there be `return` before this `reject`?

Comment (sdomagala) [reply src/middlewares/__tests__/secretsManager.js]: Yes, I must've deleted it when cleaning up.
Thanks!

File: docs/middlewares.md [hunk 2]
```diff
@@ -487,6 +487,7 @@ For each secret, you also provide the name under which its value should be added
 - `secrets` (object) : Map of secrets to fetch from Secrets Manager, where the key is the destination, and value is secret name in Secrets Manager.
   Example: `{secrets: {RDS_LOGIN: 'dev/rds_login'}}`
 - `awsSdkOptions` (object) (optional): Options to pass to AWS.SecretsManager class constructor.
+- `shouldFail` (boolean) (optional): Defaults to `false`. Set it to `true` if you want your lambda to fail in case call to AWS Secrets Manager fails (secrets don't exist or internal error)
```
Comment (vladgolubev) [inline docs/middlewares.md]: Maybe name it more explicitly to better convey the intent?

Like `throwOnMissingParam` or `ignoreMissingParams`

File: src/middlewares/__tests__/secretsManager.js [hunk 3]
```diff
@@ -217,6 +218,56 @@ describe('🔒 SecretsManager Middleware', () => {
...
+    })
+    return testScenario({
+      mockResponse: {},
+      middlewareOptions: {
+        secrets: {
+          KEY_NAME: 'failed_call'
+        }
+      },
+      callbacks: [
+        (_, {context}) => {
```
Comment (vladgolubev) [inline src/middlewares/__tests__/secretsManager.js]: Unused `context` variable here?

Comment (sdomagala) [reply src/middlewares/__tests__/secretsManager.js]: eslint didn't mention it so I thought that's not an issue but sure, I'll delete it

File: docs/middlewares.md (hunk 2 above)
Comment (sdomagala) [reply docs/middlewares.md]: Yeah I couldn't think of any good name, your proposal doesn't cover errors other than missing param in store - how about `throwOnFailedCall`?

Comment (vladgolubev) [reply docs/middlewares.md]: Hmm, agree, doesn't cover throttling errors etc, so another thing I can think of is to borrow [bail](https://jestjs.io/docs/en/cli#bail) name from jest

Comment (sdomagala) [reply docs/middlewares.md]: Hmm, I think `throw` would be more descriptive here, `bail` is not so common

**Comment by sdomagala** [comment]
Thanks @vladgolubev for quick review!

File: src/middlewares/secretsManager.js [hunk 4]
```diff
@@ -57,6 +58,9 @@ module.exports = opts => {
           // if we already have a cached secrets, then reset the timestamp so we don't
           // keep retrying on every invocation which can cause performance problems
           // when there's temporary problems with Secrets Manager
+          if (options.throwOnFailedCall) {
```
Comment (theburningmonk) [inline src/middlewares/secretsManager.js]: I think we should only throw on the first request, if you have a cached value already you can continue with execution still. If we want to be smarter then we can build in some mechanism for the app developer to flush the cache - for example, by adding a function to the context object to `flushCachedSecrets`, but let's not do it in this PR.

Comment (sdomagala) [reply src/middlewares/secretsManager.js]: Hey @theburningmonk, 
I didn't thought of it, it's fixed now, lambda will fail only if first call didn't succeed

**Comment by theburningmonk** [comment]
@sdomagala thanks for making the change, LGTM!

**Comment by lmammino** [comment]
Hey Guys, sorry I have been a bit absent these days. Thanks for taking care of this and doing such a thorough review.

I'll version bump and get this merged.

**Comment by lmammino** [comment]
@sdomagala, thanks for proposing this change. I tried to version bump (to version `0.18.0`) in your branch but it doesn't let me push. can you update `package.json` and `package-lock.json` and change the version to `0.18.0` so that we can get this merged and let the CI pipeline release the new version?

Thanks again :)

**Comment by sdomagala** [comment]
Hey @lmammino, sure, on it!

**Comment by sdomagala** [comment]
@lmammino, could you update package-lock yourself? I'm using yarn exclusively and get a lot of checksum errors when going the npm way, which then result in dependencies refresh.
I'll give you write access to this repo

**Pull request merged**